lib/m2ts/caption-stream.js

```diff
@@ -402,47 +662,67 @@ Cea608Stream.prototype.flushDisplayed = function(pts) {
     .map(function(row) {
       return row.trim();
     })
-    // remove empty rows
-    .filter(function(row) {
-      return row.length;
-    })
     // combine all text rows to display in one cue
-    .join('\n');
+    .join('\n')
+    // and remove blank rows from the start and end, but not the middle
+    .replace(/^\n+|\n+$/g, '');
 
   if (content.length) {
     this.trigger('data', {
       startPts: this.startPts_,
       endPts: pts,
-      text: content
+      text: content,
+      track: this.track_
     });
   }
 };
 
+// Adds the opening HTML tag for the passed character to the caption text,
+// and keeps track of it for later closing
+Cea608Stream.prototype.addFormatting = function(pts, format) {
+  var text = '';
+  for (var i = 0; i < format.length; i++) {
+    text += '<' + format[i] + '>';
+  }
+  this.formatting_ += format;
```

Comment:
If `this.formatting_` was always an array of single characters it would make things easier later when you want to unset them because you can use the much prettier iterator functions!

```javascript
Cea608Stream.prototype.addFormatting = function(pts, format) {
  this.formatting_ = this.formatting_.concat(format);
  
  return format.reduce(function (text, format) {
    return text + '<' + format + '>';
  }, '');
};

Cea608Stream.prototype.clearFormatting = function(pts) {
  if (!this.formatting_.length) {
    return;
  }
  const text = this.formatting_.reverse().reduce(function (text, format) {
    return text + '</' + format + '>';
  }, '');
  this.formatting_ = [];
  this[this.mode_](pts, text);
};
```